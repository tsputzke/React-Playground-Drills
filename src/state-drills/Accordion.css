.button-styles {
  background-color: pink;
}