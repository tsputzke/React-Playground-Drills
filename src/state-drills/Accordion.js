import React from 'react';
import './Accordion.css'

class Acordion extends React.Component {
  static defaultProps = { sections: []};
  state = {
    currentTabIndex: null
  };
  handleButtonClick(index) {
    this.setState({
      currentTabIndex: index
    })
  }
  renderButtons() {
    return this.props.sections.map((tab, index) => (
      <li  key={index}>
        <button className="button-styles" onClick={() => this.handleButtonClick(index)}>
        {tab.title}
        </button>
      </li>
    ))
  }
  renderContent() {
    if(this.state.currentTabIndex != null) {
      const currentTab = this.props.sections[this.state.currentTabIndex]
      return (
        <p>{currentTab.content}</p>
      )
    }
  }
  render() {
    return (
      <div>
        <ul>
        {this.renderButtons()}
        </ul>
        {!!this.props.sections.length && this.renderContent()}
      </div>      
    )
  }
}

export default Acordion