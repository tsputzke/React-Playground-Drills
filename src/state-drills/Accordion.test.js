import React from 'react';
import ReactDOM from 'react-dom';
import Accordion from './Accordion';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';import { isThisTypeAnnotation, exportAllDeclaration } from '@babel/types';
;

describe('Accordian Component', () => {
  const sections = [
    {
      title: 'Section 1',
      content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
    },
    {
      title: 'Section 2',
      content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
    },
    {
      title: 'Section 3',
      content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
    },
  ]

  it('renders empty li given no section prop', () => {
    const wrapper = shallow(<Accordion />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders no section active by default', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('opens any clicked section', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('only opens one section at a time', () => {
    const wrapper = shallow(<Accordion sections={sections} />)
    wrapper.find('button').at(1).simulate('click')
    wrapper.find('button').at(2).simulate('click')
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})