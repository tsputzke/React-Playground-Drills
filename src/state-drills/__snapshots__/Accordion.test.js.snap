// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accordian Component only opens one section at a time 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        className="button-styles"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        className="button-styles"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        className="button-styles"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
  <p>
    Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?
  </p>
</div>
`;

exports[`Accordian Component opens any clicked section 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        className="button-styles"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        className="button-styles"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        className="button-styles"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
  <p>
    Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
  </p>
</div>
`;

exports[`Accordian Component renders empty li given no section prop 1`] = `
<div>
  <ul />
</div>
`;

exports[`Accordian Component renders no section active by default 1`] = `
<div>
  <ul>
    <li
      key="0"
    >
      <button
        className="button-styles"
        onClick={[Function]}
      >
        Section 1
      </button>
    </li>
    <li
      key="1"
    >
      <button
        className="button-styles"
        onClick={[Function]}
      >
        Section 2
      </button>
    </li>
    <li
      key="2"
    >
      <button
        className="button-styles"
        onClick={[Function]}
      >
        Section 3
      </button>
    </li>
  </ul>
</div>
`;
